### DISCUSSION TITLE
which nodeModel return __gatsby_resolved ?

### DISCUSSION THREAD

**Discussion by hasanm95** [gatsbyjs/gatsby#36148] [Help] [answered]
runQuery nodeModel used to return __gatsby_resolved. Since now runQuery is deprecated is there any nodeModel which return __gatsby_resolved?

**Comment by LekoArts** [comment]
This is/was a private implementation detail that you shouldn't use as it's not public API. If you can explain what you want to do we can surely point to a safer public API.

**Comment by hasanm95** [comment]
I have been following [Json Langstorf's this repo](https://github.com/jlengstorf/gatsby-advanced-graphql/blob/master/gatsby-node.js) to customize my schema. But, now It is not working with the latest gatsby version.

**Comment by LekoArts** [comment, marked as answer]
You probably will want to use something like this:

```ts
const mdResolverPassthrough = (fieldName: string) => async (source, args, context, info) => {
  const type = info.schema.getType(`MarkdownRemark`)
  const mdNode = context.nodeModel.getNodeById({
    id: source.parent,
  })
  const resolver = type.getFields()[fieldName].resolve
  const result = await resolver(mdNode, args, context, info)
  return result
}

export const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] = ({ actions }): void => {
  const { createTypes, createFieldExtension } = actions

  createFieldExtension({
    name: `mdpassthrough`,
    args: {
      fieldName: `String!`,
    },
    extend({ fieldName }) {
      return {
        resolve: mdResolverPassthrough(fieldName),
      }
    },
  })

  createTypes(`#graphql
    type YourType {
      someValue: String! @mdpassthrough(fieldName: "otherValue")
    }
  `)
}
```

And then adapt it to what you want to do
